vaadin-split-layout-flow/src/main/java/com/vaadin/flow/component/splitlayout/SplitLayout.java

```diff
@@ -280,7 +283,32 @@ public Component getSecondaryComponent() {
      * @param position the relative position of the splitter, in percentages
      */
     public void setSplitterPosition(double position) {
-        double primary = Math.min(Math.max(position, 0), 100);
+        this.splitterPosition = position;
+        getUI().ifPresent(this::requestStylesUpdatesForSplitterPosition);
+    }
+
+    private void requestStylesUpdatesForSplitterPosition(UI ui) {
+        this.updateStylesRegistration = debounce(this.updateStylesRegistration,
```

Comment:
does it make sense to pass the registration by reference here? 
maybe it's late, but I think something is looks more complicated than it should
we can check next week

Reply:
Removed method.